### DISCUSSION TITLE
'Push only' mode that ignores local deletions

### DISCUSSION THREAD

**Discussion by OscarVanL** [abraunegg/onedrive#2556] [Q&A] [answered]
Hi,

I am using onedrive on a server to sync some generated pdf/xlsx files to a shared SharePoint folder, it's been doing a great job for the last 1 year 2 months. Thank you for your work and community support, it is much appreciated 🎉.

Unfortunately due to some error _on our end_, the folder we were syncing got deleted on the host machine. Of course, onedrive tried to sync this deletion up to the shared folder, but **thankfully** failed due to a Microsoft Graph error:
```
2023-12-05 09:47:41.240 onedrive Received inotify delete event from operating system .. attempting item deletion as requested
2023-12-05 09:47:41.240 onedrive [M] Local item deleted: ./ImportantDirectory
2023-12-05 09:47:41.240 onedrive Deleting item from OneDrive: ./ImportantDirectory
2023-12-05 09:47:43.290 onedrive null
2023-12-05 09:47:43.290 onedrive ERROR: Microsoft OneDrive API returned an error with the following message:
2023-12-05 09:47:43.290 onedrive   Calling Function: uploadDeleteItem()
2023-12-05 09:47:43.290 onedrive   Error Message:    HTTP request returned status code 422 (Unprocessable Entity)
2023-12-05 09:47:43.290 onedrive null
2023-12-05 09:47:43.290 onedrive   Error Reason:     The request is unprocessable because it uses too many resources
2023-12-05 09:47:43.290 onedrive   Error Timestamp:  2023-12-05T09:47:43
```

This made me realise that this is _not_ configured to do what we want here and we've misconfigured it. All I need is that new files created in a directory get synced up - a one-way operation. I certainly don't want deletions to get synced up, and I never access these files again on the server. No one changes the files on OneDrive/sharepoint, so I don't care about syncing the files down. 

I see I should probably be running the application with `onedrive --synchronize --upload-only --no-remote-delete` to have a one-way upload-only sync with deletions ignored.

However, the docs make me unsure that even this achieves what we want, it says [here](https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#performing-a-one-way-upload-sync):
> Note: The operation of 'upload only' does not request data from OneDrive about what 'other' data exists online. The client only knows about the data that 'this' client uploaded, thus any files or folders created or uploaded outside of this client will remain untouched online.

This seems to imply that if "this" client deleted the files, those deletions would be synced up too. This isn't what I want either.

Is there a way to run onedrive in an "absolutely do not delete anything under any circumstance" mode? I think I can kind-of achieve this with `classify_as_big_delete=1`, but then it means if something got deleted on "this" client then it would halt syncing, which isn't exactly what I want either. Really, I just want to upload what does exist and ignore any deletions.

Is there any way to achieve this?

**Comment by abraunegg** [comment, marked as answer]
@OscarVanL 
> This seems to imply that if "this" client deleted the files, those deletions would be synced up too. This isn't what I want either.

No - the wording *does not* say that. Please re-read.

This client will *only* know about (the local cache file) about data that this client has uploaded. 

The option `--upload-only` is just that .... it uploads only. It uploads *all* local changes to OneDrive. A change is:
* Move
* Add
* Content change
* Delete

The option `--no-remote-delete` *prevents* that last operation - so that you have essentially a 1 way archive occurring with zero deletion.

> Is there a way to run onedrive in an "absolutely do not delete anything under any circumstance" mode?

Yes - `onedrive --synchronize --upload-only --no-remote-delete` is how you achieve this .. 

If you have doubts, use `--dry-run` to confirm.

> I think I can kind-of achieve this with `classify_as_big_delete=1`, but then it means if something got deleted on "this" client then it would halt syncing, which isn't exactly what I want either.

And this would also be 100% non-productive either, as in v2.5.x this sort of configuration is prevented.

**Reply by OscarVanL** [reply]
Thank you very much! That's exactly what I needed to clarify.
